index.js

```diff
@@ -1,135 +1,35 @@
-
-    });
-
-    yargs.command('notebook <notebook>', 'Test notebook html and report feedback into rendered output', (yargs) => { }, async (argv) => {
-        await notebook(argv);
+    yargs.command('report <doc> [html]', 'Test markdown/steps file and report feedback into rendered output', (yargs) => { }, async (argv) => {
+        await docable(argv, true);
     });
 
     // Turn on help and access argv
     yargs.help().argv;
 
 })();
 
-async function notebook(argv) {
-    const notebookPath = argv.notebook.startsWith('/') ? argv.notebook : path.join(process.cwd(), argv.notebook);
-    let notebook;
-    try {
-        notebook = await fs.promises.readFile(notebookPath)
-    }
-    catch (err) {
-        console.error('Error:', err);
-        process.exit(1);
-    }
-
-    let stepper = new Steps();
-    let { conn, cells, $ } = await stepper.readNotebook(notebook);
-    const cwd = process.cwd();
-
-    const op = new Operators(conn, '.');
-
-    for (const cell of cells) {
-        let result;
-        switch (cell.type) {
-            case 'file':
-                result = await op.file(cell.content, cell.path, cell.user);
-                break;
-            case 'command':
-                result = await op.run(cell.content, cell.user, cell.persistent);
-                break;
-            default:
-                break;
-        }
-
-        console.log('results:', result);
+async function docable(argv, report) {
+    let stepper = new Stepper(argv.doc, argv.html);
+    await stepper.setup(argv.doc, argv.html);
+    const { $, results } = await stepper.run();
 
-        if (result.exitCode === 0) {
-            result.status = true;
-            stepper._setResults($(cell.elem).parent('pre'), result);
-        }
-        else {
-            result.status = false;
-            stepper._setResults($(cell.elem).parent('pre'), result);
-        }
-
-    }
-
-    await fs.promises.writeFile(path.join(path.dirname(notebookPath), 'notebook_results.html'), $.html(), { encoding: 'utf-8' });
-}
-
-async function testreport(mode, argv, options = {rendered: undefined, selector: undefined, css: undefined, textSelector: undefined})
-{
-    // documents and associated steps; connector to infrastructure provider
-    let stepper = new Steps(options.renderer, options.selector, options.css);
-
-    let {docs, conn, cwd, targets, clean, verify} = await stepper.read(argv.stepfile);
-
-    console.log(`Using cwd ${cwd}`);
-    let op = new Operators(conn, cwd,targets);
-    let sl = new Select( op, options.textSelector );
-
-    console.log(chalk`{bold \nRunning documentation tests:\n}`)
-
-    // Select/translate/perform/assert workflow
-    
-    let results_dir = path.join(path.dirname(argv.stepfile), 'docable_results');
-    if (!fs.existsSync(results_dir)) {
-        fs.mkdirSync(results_dir);
+    if (report) {
+        const reporter = new Reporter($, results);
+        let docStatus = await reporter.report();
```

Comment:
a) Shouldn't we know the overall status from the execution results instead of the reporter, which is a _display_ concern?

b) We only set the exit code if we have a `report=true`. Seems like we should be able to determine that regardless of having a particular report or not.

Reply:
resolved in 77bab1e.